pkg/dns/aws/dns.go

```diff
@@ -477,3 +479,21 @@ func (m *Provider) updateRecord(domain, zoneID, target, targetHostedZoneID, acti
 	log.Info("updated DNS record", "zone id", zoneID, "domain", domain, "target", target, "response", resp)
 	return nil
 }
+
+// route53GovCloud returns true if the route53 client is configured for the
+// AWS GovCloud partition or uses a GovCloud custom endpoint from config.
+func (m *Provider) route53GovCloud() bool {
+	switch {
+	case m.route53.PartitionID == endpoints.AwsUsGovPartitionID:
+		return true
+	case m.config.ServiceEndpoints != nil:
+		for _, ep := range m.config.ServiceEndpoints {
+			if ep.Name == Route53Service {
+				if strings.Contains(ep.URL, govCloudRoute53Region) {
+					return true
+				}
+			}
+		}
```

Comment:
Could this be simplified like the following?
```suggestion
	case strings.Contains(m.route53.Endpoint, govCloudRoute53Region):
		return true
```

Is it necessary to check both the partition and the endpoint?

What do you think about making this a function for simplicity and testability?
```go
func (m *Provider) updateRecord(domain, zoneID, target, targetHostedZoneID, action string, ttl int64) error {
// ...
	if clientEndpointIsGovCloud(&m.route53.Client.ClientInfo) {
// ...

// clientEndpointIsGovCloud returns true if the provided client info references
// a US GovCloud API endpoint.
func clientEndpointIsGovCloud(clientInfo *metadata.ClientInfo) bool {
	return strings.Contains(clientInfo.Endpoint, govCloudRoute53Region)
}
```￼